crates/wasm-encoder/src/core/code.rs

```diff
@@ -772,1547 +775,1550 @@ pub enum Instruction<'a> {
     F64x2RelaxedMax,
 }
 
-impl Instruction<'_> {
-    pub(crate) fn encode(&self, bytes: &mut Vec<u8>) {
+impl Encode for Instruction<'_> {
+    fn encode<S>(&self, sink: &mut S)
```

Comment:
This is somewhat preexisting but I think this refactor makes things a bit worse. I'm pretty wary of the `S` type parameter on `Encode` because it means this entire crate is monomorphized on use, meaning that this ends up increasing compile time for consumers. Additionally I feel like the usage of iterators and adapters has increased with this change which is quite a lot for LLVM to chew through. 

Overall I would personally prefer if the trait simply took `&mut Vec<u8>` (as many encoders did internally as as did this `Instruction::encode` method. While it's not super-optimal for everyone I think it strikes a better balance in terms of compile time and development overhead.

Reply:
Happy to make it `&mut Vec<u8>`.

Reply:
I've pushed up that change.